AppCenterData/AppCenterData/Internal/Client/MSDataOperationProxy.m

```diff
@@ -239,17 +245,21 @@ - (void)updateLocalStore:(MSTokenResult *)token
     [self.documentStore deleteWithToken:token documentId:currentCachedDocument.documentId];
   }
 
-  /*
-   * If the cached document has a create or replace pending operation, and no eTags, and if the current operation is a
-   * deletion, delete the document from the store.
-   */
+   //If the cached document has a create or replace pending operation, and no eTags, delete the document from the store.
   else if (([kMSPendingOperationCreate isEqualToString:currentCachedDocument.pendingOperation] ||
             [kMSPendingOperationReplace isEqualToString:currentCachedDocument.pendingOperation]) &&
-           !currentCachedDocument.eTag && operation && [kMSPendingOperationDelete isEqualToString:(NSString *)operation]) {
+           !currentCachedDocument.eTag) {
     MSLogInfo([MSData logTag], @"Removing never-synced document from local storage (partition: %@, id: %@)", token.partition,
               currentCachedDocument.documentId);
     [self.documentStore deleteWithToken:token documentId:currentCachedDocument.documentId];
   }
+  
+  // If operation is delete, delete the document from the local storage.
+  else if ([kMSPendingOperationDelete isEqualToString:(NSString *)operation]) {
```

Comment:
This might prevent offline delete to work?

When offline, we want a deletion to place a record in the local document store.

Reply:
valid point, fixing